### ISSUE TITLE
[BUG] System.NullReferenceException: Object reference not set to an instance of an object. on start

### ISSUE THREAD

**Issue by widwing** [ApacheTech-VintageStory-Mods/NeverWinter#1]
**Describe the Bug:**
On start I am getting: 
```
25.11.2023 19:02:20 [Error] [neverwinter] An exception was thrown when trying to start the mod:
System.NullReferenceException: Object reference not set to an instance of an object.
at Gantry.Services.FileSystem.FileSystemService..ctor(FileSystemServiceOptions options)
at Gantry.Services.FileSystem.DependencyInjection.GantryDependencyInjectionExtensions.AddFileSystemService(IServiceCollection services, Action`1 options)
at ApacheTech.VintageMods.NeverWinter.Program.ConfigureUniversalModServices(IServiceCollection services, ICoreAPI api)
at Gantry.Core.DependencyInjection.ModHost.<>c__DisplayClass6_0.b__1(IServiceCollection ioc)
at ApacheTech.Common.Extensions.System.ObjectExtensions.With[T](T item, Action`1 work)
at Gantry.Core.DependencyInjection.ModHost.BuildServerHost(ICoreServerAPI sapi)
at Gantry.Core.DependencyInjection.ModHost.StartPreUniversal(ICoreAPI api)
at Gantry.Core.ModSystems.Generic.ModSystemBase.StartPre(ICoreAPI api)
at Vintagestory.Common.ModLoader.TryRunModPhase(Mod mod, ModSystem system, ICoreAPI api, ModRunPhase phase) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Common\API\ModLoader.cs:line 618
25.11.2023 19:02:20 [Error] Failed to run mod phase Pre for mod ApacheTech.VintageMods.NeverWinter.Program
```

Version of vintage story: 1.18.15
OS: Ubuntu 22.04.2 LTS

**Comment by ApacheTech** [comment]
Is this on the client, or the server? What is the title of the log file, where that error occurs? You'll see either `server-main.txt` or `client-main.txt`.

This error usually means you have some corrupted settings. The first fix would be to goto `%APPDATA%\VintagestoryData\ModData\ApacheTech Solutions`, (whatever the linux is for that) and remove the `neverwinter` folder. This will reset the settings for the mod.

**Comment by widwing** [comment]
on the server on start. The file is `server-main.txt`

`ApacheTech Solutions`  was empty. deleting the folder and restarting the server didn't help, unfortunately:(

**Comment by ApacheTech** [comment]
Could you test one of my other mods on your server, for instance, Knapster? I'm concerned that Linux might not like spaces in the folder names. If that is the case, then Knapster will fail with roughly the same error.

**Comment by widwing** [comment]
Tried Knapster. same error. very likely you are right about the name

**Comment by ApacheTech** [comment]
Ok, thanks. Leave it with me. I'll have some time over the weekend to sort it out. I'll need to re-release all of my mods, with a patch. And I need to copy the current files over to the new directories, so people don't lose settings. Thank you for bringing this to my attention.

I use the same MDK for all of my mods, so if a bug is found in the MDK, it affects all my mods. I'm surprised this hasn't been picked up before. I know this is an issue with mod folders as well, if you install the game in a folder called "Vintage Story".

**Comment by ApacheTech** [comment]
I've released 1.4.0 of the mod, that should fix this issue.

**Comment by widwing** [comment]
Hi. sorry, couldn't test earlier. I just tested 1.4.0, and it has same problem. it still creates the folder with a space in it
